File: python/链表/反转链表.py
def reverseListNOde(head):
    # 双指针法
    if head is None or head.next is None:
        return head

    cur = head
    pre = None
    while cur:
        temp = cur.next
        cur.next = pre
        pre = cur
        cur = temp
    return pre

File: golang/链表/反转链表.go
package main

func reverseListNode(head *ListNode) *ListNode {
	if head == nil || head.Next == nil {
		return head
	}

	cur := head
	per := &ListNode{}
	for cur.Next != nil {
		tmp := cur.Next
		cur.Next = per
		per = cur
		cur = tmp
	}

	return per
}
